CONTRIBUTING.md

```diff
@@ -79,3 +81,132 @@ We use [towncrier](https://github.com/twisted/towncrier) for managing our change
+
+```py
+    def pins_from(self, channel_id: Snowflake) -> Response[List[message.Message]]:
+        return self.request(Route("GET", "/channels/{channel_id}/pins", channel_id=channel_id))
+```
+
+This is the basic model that all API request methods follow. Define the `Route`, provide the major parameters (in this example `channel_id`), then return a call to `self.request()`, and finally typehints.
+
+The `Response[]` part in this typehint is referring to `self.request`, as the important thing here is that `pins_from` is **not** a coroutine. Rather, `pins_from` does preprocessing and `self.request` does the actual work. The result from `pins_from` is awaited by `disnake.abc.Messageable.pins`.
+
+The Route class is how all routes are processed internally. This, along with `self.request` makes it possible to properly handle all ratelimits, and rarely encounter 429s. This is why `channel_id` is provided as a kwarg to `Route`.
+
+#### Writing Documentation
+
+While a new feature can be useful, it requires documentation to be useful too. When updating a class or method, we ask that you use
+[Sphinx directives](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionadded) in the docstring to note when it was added or updated, and what about it was updated.
+
+For example, here is the docstring for `pins()`:
+
+```py
+      """|coro|
+
+      Retrieves all messages that are currently pinned in the channel.
+
+      .. note::
+
+          Due to a limitation with the Discord API, the :class:`.Message`
+          objects returned by this method do not contain complete
+          :attr:`.Message.reactions` data.
+
+      Raises
+      ------
+      HTTPException
+          Retrieving the pinned messages failed.
+
+      Returns
+      -------
+      List[:class:`.Message`]
+          The messages that are currently pinned.
+      """
+```
+
+If we were to add a new parameter to this method, a few things would need to be added to this docstring. Lets pretend we're adding a parameter, ``oldest_first``.
+
+We use NumPy style docstrings parsed with Sphinx's Napoleon extension, the primary documentation for these docstrings can be found [here](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html).
+
+```py
+      """
+      ...
+
+      Parameters
+      ----------
+      oldest_first: bool
+          Whether to order the result by the oldest or newest pins first.
+
+          .. versionadded:: 2.9
+
+      ...
+      """
+```
+
+It is important that the section header comes **after** any description and admonitions that exist, as it will stop the parsing of the description.
+
+It is also desired to put a ``.. versionchanged:: 2.9`` directly below the section header with which parameter was changed, if the function has multiple arguments.
```

Comment:
Not sure about this, this doesn't match what we've been doing previously.. one directive is enough.

Reply:
https://github.com/DisnakeDev/disnake/pull/601/commits/afe032dce0ca1800f503654063f908d093c16f58